### ISSUE TITLE
`Script location could not be found!` on zsh 5.0.7

### ISSUE THREAD

**Issue by protist** [bhilburn/powerlevel9k#264]
I have powerlevel9k (via zim) installed fine on my laptop, which runs Arch Linux and zsh 5.2. I'm trying to install it on my Raspberry Pi server, running Raspbian and zsh 5.0.7. However, when I start a new shell I'm presented with `Script location could not be found!`.

I looked in your code, and I think the problem is that `$0` cannot be reassigned on this version of zsh. As a test, I created the following script (`./foo`).
```
#!/usr/bin/env zsh

echo $0
0=bar
echo $0
```
When run on my Arch zsh 5.2, `$0` is reassigned fine.
```
$ ./foo
./foo
bar
```
However, on my Raspbian zsh 5.0.7, it doesn't reassign.
```
$ ./foo
./foo
./foo
```

As a workaround, I modified `.zim/modules/prompt/external-themes/powerlevel9k/powerlevel9k.zsh-theme` to suppress the error, and hardcode the `filename` as follows.

```
--- powerlevel9k.zsh-theme.orig	2016-05-18 16:30:03.356684456 +1000
+++ powerlevel9k.zsh-theme	2016-05-18 16:29:57.466689508 +1000
@@ -55,10 +55,8 @@
 elif [[ -f $0 ]]; then
   # Script is a file
   filename="$0"
-else
-  print -P "%F{red}Script location could not be found!%f"
-  return 1
 fi
+filename="$POWERLEVEL9K_INSTALLATION_PATH"
 script_location="$(dirname $filename)"
 
 ################################################################
```

This now loads fine in zsh 5.0.7.

**Comment by dritter** [comment]
Yes, we need at least ZSH 5.0.8 for the finding mechanism to work.
But we could reassign `$0` to `$POWERLEVEL9K_INSTALLATION_PATH` if that variable was set. In addition we just need to refactor the [fallback mechanism](https://github.com/bhilburn/powerlevel9k/blob/0b57ee3669dd2d6d536ecff116345a929f6b068c/powerlevel9k.zsh-theme#L33-L47), so links and directories are handled as well.
That way, the warning could stay in the code. IMHO it is a good hint why things do not work correctly.

**Comment by protist** [comment]
Are you saying that zsh ≥ 5.0.8 is needed for other powerlevel9k functions? And hence having some kind of warning is important? Unfortunately, Raspbian is stuck on a fairly old version. If it's just this "finding mechanism" that relies on recent versions of zsh, then it's fairly easy to fix by using a new "normal" variable.

**Comment by dritter** [comment]
AFAIK it's just the finding mechanism. The "normal" variable should be the `POWERLEVEL9K_INSTALLATION_PATH`.
The warning could be more helpful, indeed.

**Comment by protist** [comment]
Ah okay, I see. Seems like a relatively straightforward fix then, hopefully. Cheers.

**Comment by bhilburn** [comment]
Fixed my #266 merge.
